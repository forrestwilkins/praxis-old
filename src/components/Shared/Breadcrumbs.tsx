import { useReactiveVar } from "@apollo/client";
import { Breadcrumbs, Typography } from "@material-ui/core";
import Link from "next/link";

import { breadcrumbsVar } from "../../apollo/client/localState";

const SharedBreadcrumbs = () => {
  const breadcrumbs = useReactiveVar(breadcrumbsVar);

  if (breadcrumbs.length)
    return (
      <Breadcrumbs>
        {breadcrumbs.map(({ label, href }) => {
          if (href)
            return (
              <Link href={href} key={href}>
                <a>
                  <Typography>{label}</Typography>
                </a>
              </Link>
            );
          return (
            <Typography color="primary" key={label}>
              {label}
            </Typography>
          );
        })}
      </Breadcrumbs>
    );

  return null;
};

export default SharedBreadcrumbs;
