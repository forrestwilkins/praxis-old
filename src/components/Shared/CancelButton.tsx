import {
  Button as MUIButton,
  ButtonProps,
  createStyles,
  withStyles,
} from "@material-ui/core";
import Messages from "../../utils/messages";

const Button = withStyles(() =>
  createStyles({
    root: {
      fontFamily: "Inter Bold",
      letterSpacing: "0.2px",
      textTransform: "none",
      minWidth: 80,
    },
  })
)(MUIButton);

const CancelButton = (props: ButtonProps) => {
  return (
    <Button variant="contained" color="secondary" {...props}>
      {props.children ? props.children : Messages.actions.cancel()}
    </Button>
  );
};

export default CancelButton;
