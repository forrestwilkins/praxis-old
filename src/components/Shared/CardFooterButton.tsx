import {
  ButtonProps,
  createStyles,
  makeStyles,
  Button,
} from "@material-ui/core";

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      textTransform: "none",
      paddingLeft: 12,
      paddingRight: 12,
    },
  })
);

const CardFooterButton = (props: ButtonProps) => {
  const classes = useStyles();

  return (
    <Button color="primary" classes={classes} {...props}>
      {props.children}
    </Button>
  );
};

export default CardFooterButton;
