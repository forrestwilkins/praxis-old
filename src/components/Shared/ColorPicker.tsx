import { useState } from "react";
import { CirclePicker, ColorResult } from "react-color";
import { ArrowForwardIos } from "@material-ui/icons";

import styles from "../../styles/Shared/ColorPicker.module.scss";
import Messages from "../../utils/messages";

interface Props {
  label: string;
  color: string;
  onChange: (color: ColorResult) => void;
}

const ColorPicker = ({ label, color, onChange }: Props) => {
  const [open, setOpen] = useState<boolean>(false);

  return (
    <>
      <div
        onClick={() => setOpen(!open)}
        className={styles.field}
        role="button"
        tabIndex={0}
      >
        <div className={styles.label}>{label}</div>
        <div className={styles.color}>
          <div
            style={{ backgroundColor: color }}
            className={styles.colorSquare}
          ></div>
          <div className={styles.colorHex}>{color}</div>
          <ArrowForwardIos
            color="inherit"
            fontSize="small"
            style={{ marginTop: 3 }}
          />
        </div>
      </div>

      {open && (
        <div className={styles.colorPickerWrapper}>
          <div className={styles.pickPrompt}>
            {Messages.actions.pickColor()}
          </div>
          <CirclePicker onChangeComplete={onChange} />
        </div>
      )}
    </>
  );
};

export default ColorPicker;
