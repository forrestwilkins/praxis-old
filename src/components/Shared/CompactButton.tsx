import {
  ButtonProps,
  createStyles,
  makeStyles,
  Button,
  Theme,
} from "@material-ui/core";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: 0,
      minWidth: "initial",
      borderRadius: 4,
    },
    textPrimary: {
      "&:hover": {
        backgroundColor: theme.palette.background.paper,
      },
    },
  })
);

const CompactButton = (props: ButtonProps) => {
  const classes = useStyles();
  return (
    <Button color="primary" classes={classes} {...props}>
      {props.children}
    </Button>
  );
};

export default CompactButton;
