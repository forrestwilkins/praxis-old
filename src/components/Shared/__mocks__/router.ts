export const useRouter = jest.spyOn(require("next/router"), "useRouter");

export const mockNextUseRouter = (pathname: string) => {
  useRouter.mockImplementation(() => ({
    pathname,
    query: {},
    asPath: "",
    push: async () => true,
    prefetch: async () => undefined,
  }));
};
