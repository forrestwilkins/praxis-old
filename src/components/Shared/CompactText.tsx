import { useState } from "react";
import { truncate } from "lodash";
import { TruncationSizes } from "../../constants/common";
import Messages from "../../utils/messages";
import { WHITE } from "../../styles/Shared/theme";

interface Props {
  text: string;
}

const CompactText = ({ text }: Props) => {
  const [showMore, setShowMore] = useState(false);

  if (showMore || text.length <= TruncationSizes.Large) return <>{text}</>;

  return (
    <>
      {truncate(text, {
        length: TruncationSizes.Large,
      }) + " "}
      <span
        onClick={() => setShowMore(true)}
        role="button"
        tabIndex={0}
        style={{ cursor: "pointer", color: WHITE }}
      >
        {Messages.actions.seeMore()}
      </span>
    </>
  );
};

export default CompactText;
