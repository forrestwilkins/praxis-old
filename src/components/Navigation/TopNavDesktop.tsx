import { useEffect, useState } from "react";
import { useRouter } from "next/router";
import Link from "next/link";
import { ArrowDropDown } from "@material-ui/icons";
import {
  AppBar,
  Toolbar,
  makeStyles,
  createStyles,
  IconButton,
  Button,
  Typography,
} from "@material-ui/core";

import Messages from "../../utils/messages";
import { useCurrentUser, useUserById } from "../../hooks";
import styles from "../../styles/Navigation/TopNavDesktop.module.scss";
import { NavigationPaths, ResourcePaths } from "../../constants/common";
import UserAvatar from "../Users/Avatar";
import TopNavDropdown from "./TopNavDropdown";
import SearchBar from "../Shared/SearchBar";
import { redeemedInviteToken } from "../../utils/clientIndex";

const useStyles = makeStyles(() =>
  createStyles({
    toolbar: {
      minHeight: 60,
    },
  })
);

const TopNavDesktop = () => {
  const currentUser = useCurrentUser();
  const [menuAnchorEl, setMenuAnchorEl] = useState<null | HTMLElement>(null);
  const [inviteToken, setInviteToken] = useState<string | null>();
  const [user] = useUserById(currentUser?.id);
  const classes = useStyles();
  const router = useRouter();

  useEffect(() => {
    setInviteToken(redeemedInviteToken());
  }, []);

  const handleMenuButtonClick = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    setMenuAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setMenuAnchorEl(null);
  };

  return (
    <AppBar style={{ boxShadow: "none" }}>
      <Toolbar classes={{ root: classes.toolbar }}>
        <div className={styles.appBarBrand}>
          {router.asPath === NavigationPaths.Home ? (
            <span onClick={() => router.reload()} role="button" tabIndex={0}>
              {Messages.brand()}
            </span>
          ) : (
            <Link href={NavigationPaths.Home} passHref>
              {Messages.brand()}
            </Link>
          )}
        </div>

        <SearchBar />

        <div className={styles.appBarButtons}>
          {currentUser && user && (
            <Link href={`${ResourcePaths.User}${user.name}`} passHref>
              <Button color="primary" style={{ textTransform: "none" }}>
                <UserAvatar user={user} small withoutLink />
                <Typography color="primary" style={{ marginLeft: 12 }}>
                  {user.name}
                </Typography>
              </Button>
            </Link>
          )}

          {!currentUser && (
            <Link href={NavigationPaths.LogIn}>
              <a>
                <Button color="primary">
                  {Messages.users.actions.logIn()}
                </Button>
              </a>
            </Link>
          )}

          {!currentUser && inviteToken && (
            <Link href={NavigationPaths.SignUp}>
              <a>
                <Button color="primary">
                  {Messages.users.actions.signUp()}
                </Button>
              </a>
            </Link>
          )}

          {currentUser && (
            <>
              <IconButton onClick={handleMenuButtonClick}>
                <ArrowDropDown color="primary" />
              </IconButton>

              <TopNavDropdown
                anchorEl={menuAnchorEl}
                handleClose={handleClose}
              />
            </>
          )}
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default TopNavDesktop;
