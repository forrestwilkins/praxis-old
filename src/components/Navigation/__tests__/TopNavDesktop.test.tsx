import { render, RenderResult, waitFor } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { mocked } from "ts-jest/utils";

import Messages from "../../../utils/messages";
import { TypeNames } from "../../../constants/common";
import TopNavDesktop from "../TopNavDesktop";
import { useCurrentUser } from "../../../hooks";
import { redeemedInviteToken } from "../../../utils/clientIndex";
import { mockNextUseRouter } from "../../Shared/__mocks__";

jest.mock("../../../hooks", () => ({
  useCurrentUser: jest.fn(),
}));

jest.mock("../../../utils/clientIndex", () => ({
  redeemedInviteToken: jest.fn(),
}));

const mockUserName = "mockUserName";
const mockCurrentUser: CurrentUser = {
  id: "1",
  name: mockUserName,
  email: "test@email.com",
  isAuthenticated: true,
  __typename: TypeNames.CurrentUser,
};
const mockInviteToken = "mockInviteToken";

const renderTopNavDesktop = async (): Promise<RenderResult> => {
  mockNextUseRouter("/");
  return render(
    <MockedProvider>
      <TopNavDesktop />
    </MockedProvider>
  );
};

describe("Top Navigation for Desktop", () => {
  it.skip("should render TopNavDesktop component with log in when user is logged out", async () => {
    mocked(useCurrentUser).mockReturnValueOnce(undefined);
    mocked(redeemedInviteToken).mockReturnValueOnce(null);

    const { getByText, container } = await renderTopNavDesktop();
    const logIn = getByText(Messages.users.actions.logIn());

    await waitFor(() => expect(logIn).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });

  it.skip("should render TopNavDesktop component with sign up and log in when user is logged out and has been invited", async () => {
    mocked(useCurrentUser).mockReturnValueOnce(undefined);
    mocked(redeemedInviteToken).mockReturnValueOnce(mockInviteToken);

    const { getByText, container } = await renderTopNavDesktop();
    const signUp = getByText(Messages.users.actions.signUp());
    const logIn = getByText(Messages.users.actions.logIn());

    await waitFor(() => expect(signUp).toBeInTheDocument());
    await waitFor(() => expect(logIn).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });

  it.skip("should render TopNavDesktop component with profile and log out when user is logged in", async () => {
    mocked(useCurrentUser).mockReturnValueOnce(mockCurrentUser);

    const { getByText, container } = await renderTopNavDesktop();
    const logOut = getByText(Messages.users.actions.logOut());
    const profile = getByText(mockUserName);

    await waitFor(() => expect(profile).toBeInTheDocument());
    await waitFor(() => expect(logOut).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });
});
