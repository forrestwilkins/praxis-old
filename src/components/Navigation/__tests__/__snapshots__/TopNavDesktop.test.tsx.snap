// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Navigation for Desktop should render TopNavDesktop component with log in when user is logged out 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/login"
        >
          Log in
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Top Navigation for Desktop should render TopNavDesktop component with profile and log out when user is logged in 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/mockUserName"
        >
          mockUserName
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <div
          role="button"
          tabindex="0"
        >
          <a
            class="navbarItemText"
          >
            Log out
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Top Navigation for Desktop should render TopNavDesktop component with sign up and log in when user is logged out and has been invited 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/login"
        >
          Log in
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/signup"
        >
          Sign up
        </a>
      </div>
    </div>
  </nav>
</div>
`;
